lib/src/arithmetic/compound_adder.dart

```diff
@@ -224,10 +232,10 @@ class CarrySelectOnesComplementCompoundAdder extends CompoundAdder {
   /// [CarrySelectCompoundAdder.splitSelectAdderAlgorithmSingleBlock],
   CarrySelectOnesComplementCompoundAdder(super.a, super.b,
       {Adder Function(Logic, Logic, {Logic? carryIn}) adderGen =
-          ParallelPrefixAdder.new,
+          NativeAdder.new,
       Logic? subtractIn,
-      Logic? carryOut,
-      Logic? carryOutP1,
+      bool outputCarryOut = false,
```

Comment:
doc comment explaining what these bool arguments do?  maybe add doc comment to the outputs themselves explaining that they are only generated when those bools are set?

some other components (e.g. the FIFO) had used the the terminology of `generateNameOfPort` rather than `outputNameOfPort`.  I think I'd slightly vote for "generate" since it's not overloaded with the name of an "output", but don't feel super strongly.  Consistency would be good, at least.  What do you think?

Reply:
Agreed.  generate is a better prefix.  I tried to find all my uses of this pattern and convert to generate<portName> and document.